src/Sulu/Bundle/CustomUrlBundle/Controller/CustomUrlController.php

```diff
@@ -0,0 +1,214 @@
+use JMS\Serializer\SerializationContext;
+use Sulu\Component\CustomUrl\Manager\CannotDeleteCurrentRouteException;
+use Sulu\Component\DocumentManager\Exception\DocumentNotFoundException;
+use Sulu\Component\Rest\Exception\RestException;
+use Sulu\Component\Rest\RequestParametersTrait;
+use Sulu\Component\Rest\RestController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Provides rest api for custom-urls.
+ *
+ * @RouteResource("custom-urls")
+ */
+class CustomUrlController extends RestController
+{
+    use RequestParametersTrait;
+
+    private static $relationName = 'custom-urls';
+
+    /**
+     * Returns a list of custom-urls.
+     *
+     * @param string $webspaceKey
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function cgetAction($webspaceKey, Request $request)
+    {
+        // TODO pagination
+
+        $result = $this->get('sulu_custom_urls.manager')->readList($webspaceKey, $this->getLocale($request));
+
+        $list = new RouteAwareRepresentation(
+            new CollectionRepresentation($result, self::$relationName),
+            'cget_webspace_custom-urls',
+            array_merge($request->request->all(), ['webspaceKey' => $webspaceKey])
+        );
+
+        return $this->handleView($this->view($list));
+    }
+
+    /**
+     * Returns a single custom-url identified by uuid.
+     *
+     * @param string $webspaceKey
+     * @param string $uuid
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function getAction($webspaceKey, $uuid, Request $request)
+    {
+        try {
+            $document = $this->get('sulu_custom_urls.manager')->read($uuid, $this->getLocale($request));
+        } catch (DocumentNotFoundException $ex) {
+            return $this->handleView($this->view(null, 404));
+        }
+
+        // FIXME without this target-document will not be loaded (for serialization)
+        if (null !== $document->getTarget()) {
+            $document->getTarget()->getTitle();
+        }
+
+        return $this->handleView(
+            $this->view($document)->setSerializationContext(
+                SerializationContext::create()->setGroups(['defaultCustomUrl', 'smallPage', 'fullRoute'])
+            )
+        );
+    }
+
+    /**
+     * Create a new custom-url object.
+     *
+     * @param string $webspaceKey
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function postAction($webspaceKey, Request $request)
+    {
+        try {
+            $document = $this->get('sulu_custom_urls.manager')->create(
+                $webspaceKey,
+                $request->request->all(),
+                $this->getLocale($request)
+            );
+            $this->get('sulu_document_manager.document_manager')->flush();
+        } catch (RestException $ex) {
+            return $this->handleView($this->view($ex->toArray(), 400));
+        }
```

Comment:
Same as above, handled by the FOSRestBundle now. Or add the exact exception to the configuration if it is not handled currently.